tests/ui/nll/ty-outlives/projection-where-clause-none.stderr

```diff
@@ -1,16 +1,13 @@
-error[E0309]: the parameter type `T` may not live long enough
+error[E0309]: the associated type `<T as MyTrait<'_>>::Output` may not live long enough
   --> $DIR/projection-where-clause-none.rs:14:5
    |
 LL | fn foo<'a, T>() -> &'a ()
-   |        -- the parameter type `T` must be valid for the lifetime `'a` as defined here...
+   |        -- the associated type `<T as MyTrait<'_>>::Output` must be valid for the lifetime `'a` as defined here...
 ...
 LL |     bar::<T::Output>()
-   |     ^^^^^^^^^^^^^^^^ ...so that the type `T` will meet its required lifetime bounds
+   |     ^^^^^^^^^^^^^^^^ ...so that the type `<T as MyTrait<'_>>::Output` will meet its required lifetime bounds
```

Comment:
Hm... I'll see if there's some missing call to map `'_` back to `'a`. If not, then I'm inclined to still land this change even if it's slightly worse for exactly 1 UI test...